From 2a2dcfd6658f016653f5f8a0e1f05aa44ed76bd3 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 7 Jun 2024 16:50:57 +0200
Subject: [PATCH 0921/1695] dts: stm32: define RIFSC as an access controller
 for stm32mp25 boards

RIFSC is a firewall controller. Add the access-controllers property to
all RIFSC sub-nodes. Also add the "simple-bus" compatible for backward
compatibility and "#access-controllers-cells" to the RIFSC node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8bb5e4edfcf9ec4d474934665b9cf31f5348416a dts: stm32: define RIFSC as an access controller for stm32mp25 boards]
Change-Id: I0d7a5bc9725fc6366358f82b66ff7910ecd47ce7
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/386445
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi  | 31 ++++++++++++++++++++++++++++--
 core/arch/arm/dts/stm32mp25xc.dtsi |  3 +++
 core/arch/arm/dts/stm32mp25xf.dtsi |  3 +++
 3 files changed, 35 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 0389c1eb3..f6929fbbd 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -256,11 +256,12 @@
 			};
 		};
 
-		rifsc: rifsc@42080000 {
-			compatible = "st,stm32mp25-rifsc";
+		rifsc: bus@42080000 {
+			compatible = "st,stm32mp25-rifsc", "simple-bus";
 			reg = <0x42080000 0x1000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
+			#access-controller-cells = <2>;
 
 			lptimer1: timer@40090000 {
 				compatible = "st,stm32mp25-lptimer";
@@ -270,6 +271,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM1_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -303,6 +305,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM2_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -333,6 +336,7 @@
 				reg = <0x400e0000 0x400>;
 				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART2_ID 0>;
 				status = "disabled";
 			};
 
@@ -341,6 +345,7 @@
 				reg = <0x400f0000 0x400>;
 				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART3_ID 0>;
 				status = "disabled";
 			};
 
@@ -349,6 +354,7 @@
 				reg = <0x40100000 0x400>;
 				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART4_ID 0>;
 				status = "disabled";
 			};
 
@@ -357,6 +363,7 @@
 				reg = <0x40110000 0x400>;
 				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART5_ID 0>;
 				status = "disabled";
 			};
 
@@ -368,6 +375,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C1_ID 0>;
 				status = "disabled";
 			};
 
@@ -379,6 +387,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C2_ID 0>;
 				status = "disabled";
 			};
 
@@ -390,6 +399,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C3_ID 0>;
 				status = "disabled";
 			};
 
@@ -401,6 +411,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C4_ID 0>;
 				status = "disabled";
 			};
 
@@ -412,6 +423,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C5_ID 0>;
 				status = "disabled";
 			};
 
@@ -423,6 +435,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C6_ID 0>;
 				status = "disabled";
 			};
 
@@ -434,6 +447,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C7_ID 0>;
 				status = "disabled";
 			};
 
@@ -442,6 +456,7 @@
 				reg = <0x40220000 0x400>;
 				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART6_ID 0>;
 				status = "disabled";
 			};
 
@@ -450,6 +465,7 @@
 				reg = <0x402c0000 0x400>;
 				interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART9>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART9_ID 0>;
 				status = "disabled";
 			};
 
@@ -458,6 +474,7 @@
 				reg = <0x40330000 0x400>;
 				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART1_ID 0>;
 				status = "disabled";
 			};
 
@@ -466,6 +483,7 @@
 				reg = <0x40370000 0x400>;
 				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART7_ID 0>;
 				status = "disabled";
 			};
 
@@ -474,6 +492,7 @@
 				reg = <0x40380000 0x400>;
 				interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART8>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART8_ID 0>;
 				status = "disabled";
 			};
 
@@ -483,6 +502,7 @@
 				interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_HASH>;
 				resets = <&rcc HASH_R>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_HASH_ID 0>;
 				status = "disabled";
 			};
 
@@ -492,6 +512,7 @@
 				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG>;
 				clock-names = "rng_clk", "rng_hclk";
 				resets = <&rcc RNG_R>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_RNG_ID 0>;
 			};
 
 			iwdg1: watchdog@44010000 {
@@ -500,6 +521,7 @@
 				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG1_ID 0>;
 				status = "disabled";
 			};
 
@@ -509,6 +531,7 @@
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID 0>;
 				status = "disabled";
 			};
 
@@ -520,6 +543,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C8_ID 0>;
 				status = "disabled";
 			};
 
@@ -531,6 +555,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM3_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -564,6 +589,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM4_ID 0>;
 				status = "disabled";
 
 				counter {
@@ -597,6 +623,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM5_ID 0>;
 				status = "disabled";
 
 				counter {
diff --git a/core/arch/arm/dts/stm32mp25xc.dtsi b/core/arch/arm/dts/stm32mp25xc.dtsi
index 16629ff9b..33930c9f4 100644
--- a/core/arch/arm/dts/stm32mp25xc.dtsi
+++ b/core/arch/arm/dts/stm32mp25xc.dtsi
@@ -11,6 +11,7 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID 0>;
 		status = "disabled";
 	};
 
@@ -21,6 +22,7 @@
 		clocks = <&rcc CK_BUS_SAES>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID 0>;
 		status = "disabled";
 	};
 
@@ -31,6 +33,7 @@
 		clocks = <&rcc CK_BUS_PKA>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID 0>;
 		status = "disabled";
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp25xf.dtsi b/core/arch/arm/dts/stm32mp25xf.dtsi
index df0cbc204..476679eb0 100644
--- a/core/arch/arm/dts/stm32mp25xf.dtsi
+++ b/core/arch/arm/dts/stm32mp25xf.dtsi
@@ -11,6 +11,7 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID 0>;
 		status = "disabled";
 	};
 
@@ -21,6 +22,7 @@
 		clocks = <&rcc CK_BUS_SAES>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID 0>;
 		status = "disabled";
 	};
 
@@ -31,6 +33,7 @@
 		clocks = <&rcc CK_BUS_PKA>,  <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID 0>;
 		status = "disabled";
 	};
 };
-- 
2.47.2

